internal/extproc/backendauth/aws.go

```diff
@@ -115,7 +122,12 @@ func (a *awsHandler) Do(ctx context.Context, requestHeaders map[string]string, h
 	// https://github.com/envoyproxy/envoy/blob/60b2b5187cf99db79ecfc54675354997af4765ea/source/extensions/filters/http/ext_proc/processor_state.cc#L180-L183
 	req.ContentLength = -1
 
-	err = a.signer.SignHTTP(ctx, a.credentials, req,
+	credentials, err := a.credentialsProvider.Retrieve(ctx)
```

Comment:
how costly is this `credentialsProvider.Retrieve` is my next question.. Previously, aws secrets are rotated on OIDC credentials rotation on the controller side and this newHandler was called on the update.

One another way to resolve this is to return "RequeueAfter" in this case as well 
https://github.com/envoyproxy/ai-gateway/blob/8ee7ae12c3a4534a615d704779e00ef0fb70990f/internal/controller/backend_security_policy.go#L273

which eventually propagates to the ExtProc's secret update hence the newHandler will be called. Maybe we can requeue after 1 minute or so assuming any pod will have more than 1 minute ttl on the session token. 

wdyt?